codes/c/chapter_array_and_linkedlist/list.c

```diff
@@ -0,0 +1,187 @@
+/**
+ * File: list.c
```

Comment:
"my_list.c" would be a better name for the file, because these containers are not recommended to be used in other files.